UT4MasterServer/Services/StatisticsService.cs

```diff
@@ -249,6 +249,125 @@ public async Task CreateAccountStatisticsAsync(EpicID accountID, OwnerType _, St
 		await UpdateAllTimeAccountStatisticsAsync(newStatistic);
 	}
 
+	/// <summary>
+	/// This method merges multiple statistics into one. It looks only for daily non-flagged statistics to merge that are at least 7 days old.
+	/// </summary>
+	/// <returns></returns>
+	public async Task MergeOldStatisticsAsync()
+	{
+		var dateBefore = DateTime.UtcNow.AddDays(-7).Date;
+
+		logger.LogInformation("Merging statistics older than: {Date}.", dateBefore);
+
+		var filter = Builders<Statistic>.Filter.Eq(f => f.Window, StatisticWindow.Daily) &
+					 Builders<Statistic>.Filter.Lt(f => f.CreatedAt, dateBefore) &
+					 Builders<Statistic>.Filter.Exists(f => f.Flagged, false);
+
+		var statistics = await statisticsCollection.Find(filter).ToListAsync();
+
+		if (!statistics.Any()) return;
+
+		var statisticsGrouped = statistics
+			.GroupBy(g => new { g.AccountID, g.CreatedAt.Date })
+			.Select(s => new
+			{
+				AccountId = s.Key.AccountID,
+				CreatedAt = s.Key.Date,
+				Statistics = s.ToList(),
+				Count = s.Count(),
+			})
+			.ToList();
+
+		var accountIds = statisticsGrouped.Select(s => s.AccountId).ToArray();
+		var dates = statisticsGrouped.Select(s => s.CreatedAt).ToArray();
+		var alreadyMergedFilter = Builders<Statistic>.Filter.Eq(f => f.Window, StatisticWindow.DailyMerged) &
+								  Builders<Statistic>.Filter.In(f => f.AccountID, accountIds) &
+								  Builders<Statistic>.Filter.In(f => f.CreatedAt, dates);
+
+		// This will fetch previously merged records so that un-flagged records can be merged with them
+		var alreadyMergedStatistics = await statisticsCollection.Find(alreadyMergedFilter).ToListAsync();
+
+		var mergedStatistics = new List<Statistic>();
+		var modifiedStatisticIds = new List<string>();
+		var statisticIdsToDelete = new List<string>();
+
+		foreach (var statisticsGroup in statisticsGrouped)
+		{
+			// Statistics that have multiple records per day should be merged together or merged with already merged statistics
+			if (statisticsGroup.Count > 1)
+			{
+				var statisticsToMerge = new List<Statistic>(statisticsGroup.Statistics);
+
+				if (alreadyMergedStatistics.FirstOrDefault(f => f.AccountID == statisticsGroup.AccountId &&
+																f.CreatedAt.Date == statisticsGroup.CreatedAt.Date) is { } alreadyMergedStatistic)
```

Comment:
Indentation

Reply:
Indentation is OK, just not displayed by GitHub correctly. You can check it by viewing the whole file:
![image](https://user-images.githubusercontent.com/12861786/213937489-c2f8b19e-a8e0-4a73-abfa-7f3c578050e0.png)